sdk_lib/sdk_container_common.sh

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+#
+# This file contains common functions used across SDK container scripts.
+
+#
+# globals
+#
+sdk_container_common_versionfile="sdk_container/.repo/manifests/version.txt"
+sdk_container_common_registry="ghcr.io/flatcar-linux"
+
+# Common "echo" function
+
+function yell() {
+    echo -e "\n###### $@ ######"
+}
+# --
+
+# Prevent the sourcetree from being used by multiple containers with different
+#  SDK release versions by use of a version lock.
+# The lock is self-cleaning and does not need to be removed in scripts calling
+# this function.
+function lock_version() {
```

Comment:
I don't understand for which use cases the lock should primarily be working but it's not race free

Reply:
The lock is used to prevent changing the git tree to a different version when a container is running (since the git tree is volume-mounted into the container). It's probably too complex and can be simplified.
It should not be racy, though - could you please elaborate?

Reply:
Two executions of this function can run in parallel up to the end where both will create the lock files

Reply:
Even one can complete totally and the other is still at the `if [ -z "$lock_owner_active" ] ; then` branch and will remove the lock again before it writes its own lock

Reply:
It also seems to be able to have two executions with the same version succeed to run and then there is only one owner, but a third execution will not know whether both have finished

Reply:
The directory is the lock, not the files.

The files give information about lock ownership (to figure out if the lock can safely be cleared)  to determine if the lock can safely be cleared. Ownership information is written only after acquiring the lock succeeded (in the worst of cases, a lock that could be cleared is not cleared).

Reply:
The directory doesn't really function as lock because it's created once and not in an atomic way that also creates the files that the if branches check for

Reply:
How about adding a comment that this is aimed for human usage and not race-free for automation. We can avoid running this in Jenkins as long as we kill the processes and remove any existing folders.

Reply:
Giving up for now on the multiple-containers-with-the-same-version feature and will make the locking simpler so it is safe :)

Reply:
Removed locking entirely since it was more of a hindrance than a help.